### ISSUE TITLE
[4.14][11][bahamut] No WiFi and SIM not recognized

### ISSUE THREAD

**Issue by voidanix** [sonyxperiadev/bug_tracker#694]
**Platform:** kumano
**Device:** bahamut
**Kernel version:** 4.14.232
**Android version:** 11
**Software binaries version:** SW_binaries_for_Xperia_Android_11_4.14_v4b_kumano.img

**Previously working on**
Stock (?)

**Description**
While porting LineageOS 18.1 with SODP sources I noticed that the WiFi did not work and the SIM was not recognized so I built AOSP like on the SODP website and found out the problem persists

The model is j8210 (single SIM)

**Symptoms**
Not being asked for SIM PIN code, WiFi toggles do not seem to do anything

**Additional context**
I attached a logcat, tried switching airplane mode on and off but it does not do anything either; modemswitcher seems to be the culprit?

[logcat.txt](https://github.com/sonyxperiadev/bug_tracker/files/6547824/logcat.txt)

**Comment by MarijnS95** [comment]
What was the issue here? Maybe the solution - even if simple - is relevant for other searchers too!

**Comment by voidanix** [comment]
Considering that the bootloader was unlocked recently on this device it is likely something happened under the hood (broke a TA partition?): IMEI shows up as unknown in settings and the symptoms are the same ones people without working baseband have.

Everything works fine on stock though (flashed via newflasher) and IMEI shows up correctly there as well.

tl;dr probably messed something up, if anybody else has the same issues on a kumano I will reopen this

**Comment by MarijnS95** [comment]
I wonder if that's caused by a newer bootloader and modem firmware/partition.

**Comment by voidanix** [comment]
So what you are saying is that it could be a `55.2.xxx` (the one that is currently on the phone, released in February for 11) vs. `55.1.xxx` firmware issue?

**Comment by voidanix** [comment]
You are apparently right: I went to XperiFirm, downloaded the `55.1.xxx` firmware for another operator/market (because for my global model it's only `55.2.xxx`), flashed the following files through newflasher:
```
bluetooth_X-FLASH-ALL-A2CD.sin
dsp_X-FLASH-ALL-A2CD.sin
fsg_X-FLASH-ALL-A2CD.sin
modemst1_X-FLASH-ALL-A2CD.sin
modemst2_X-FLASH-ALL-A2CD.sin
modem_X-FLASH-ALL-A2CD.sin
```

And automagically at boot I am asked for a PIN code for the SIM, WiFi works and so does LTE.

I'm reopening this then because there is clearly something wrong

**Comment by MarijnS95** [comment]
> So what you are saying is that it could be a `55.2.xxx` (the one that is currently on the phone, released in February for 11) vs. `55.1.xxx` firmware issue?

I'm saying that it is possibly too new, since SODP cannot redistribute modem partition contents but assumes it is compatible with whatever we have in vendor. You confirmed that thought by going back to an older, working version.

> I'm reopening this then because there is clearly something wrong

Let's get @jerpelea looped in and see if something can be bumped on the vendor side, **without breaking existing devices on older firmware**. That answer is usually: impossible - unless two odm's and/or a split in vendor (opensource repos) is made.

It is worth getting `radio` logs (`logcat -b radio`, or for convenience `logcat -b all`), right from the start (ie. run the command _before_ booting and immediately connecting the device, so that we have no `chatty: expire XXX lines`), to hopefully have some better hints at what exactly is broken/mismatched. Nothing stands out from a quick scroll-through, besides modemswitcher not being able to read active modems (the underlying systems might give more hints).

**Comment by voidanix** [comment]
[Here's the logcat -b all](https://github.com/sonyxperiadev/bug_tracker/files/6566448/logcatall.txt)

The first thing that pops out to me is the missing modem configuration, where with the `55.1.xxx` firmware this happens:
```
05-30 17:05:25.960  3306  3306 I sony-modem-switcher: ModemSwitcherWrapper: find_matching_mbn:Link to death notification successful!
05-30 17:05:25.961  3306  3306 W modem-switcher: Default modem conf not found: /oem/modem-config/modem.conf
05-30 17:05:25.961  3306  3306 W modem-switcher: Unable to find default modem in oem, checking ta
05-30 17:05:25.963  1931  1931 I RemoteVvmTaskManager: Binding to ComponentInfo{com.android.dialer/com.android.voicemail.impl.OmtpService}
05-30 17:05:25.967  3306  3306 W modem-switcher: Could not stat modem conf: S133.4, will use default
05-30 17:05:25.968  3310  3310 E PDC_UTIL: pdc_info init done
05-30 17:05:25.970  3310  3310 I PDC_UTIL: qmi_client_init succeeds for PDC
05-30 17:05:25.970  3310  3310 I MBN_UTIL: file name: /vendor/firmware_mnt/image/modem_pr/mcfg/configs/mcfg_sw/generic/common/default/non_ims/a/mcfg_sw.mbn
05-30 17:05:25.970  3310  3310 I MBN_UTIL: type_id: 0
05-30 17:05:25.970  3310  3310 I MBN_UTIL: type_id: 1
05-30 17:05:25.972  3310  3340 I PDC_UTIL: succeed to get config_list 1
05-30 17:05:25.973  3310  3310 I PDC_UTIL: Release client connection to PDC
```

It seems to me that the newer firmware does not pick any mbn whereas the older one does.

**Comment by MarijnS95** [comment]
> It seems to me that the newer firmware does not pick any mbn whereas the older one does.

It breaks well before that as PDC doesn't even seem to initialize, and/or the modem is not reachable over QMI:

```
05-30 16:52:40.211   856   856 E PDC_UTIL: qmi_client_init returned failure (-3) for PDC, retry times 0
```

**Comment by jerpelea** [comment]
please use 55.1.xxx, This release has been tested and is compatible with our odm
